src/app/machines/views/MachineList/MachineList.test.tsx

```diff
@@ -705,4 +714,106 @@ describe("MachineList", () => {
     expect(fetches).toHaveLength(2);
     expect(fetches[fetches.length - 1].payload.params.page_number).toBe(2);
   });
+
+  it("shows a warning notification if not all controllers are configured with Vault", async () => {
+    const controllers = [
+      controllerFactory({
+        system_id: "abc123",
+        vault_configured: true,
+        node_type: NodeType.REGION_CONTROLLER,
+      }),
+      controllerFactory({
+        system_id: "def456",
+        vault_configured: false,
+        node_type: NodeType.REGION_AND_RACK_CONTROLLER,
+      }),
+    ];
+    state.controller.items = controllers;
+
+    renderWithBrowserRouter(
+      <MachineList searchFilter="" setSearchFilter={jest.fn()} />,
+      { wrapperProps: { state } }
+    );
+
+    expect(screen.getByTestId("vault-notification")).toHaveTextContent(
```

Comment:
Notifications have a `Incomplete Vault integration` title which can be used for querying instead of a test id.

Reply:
See above